src/components/ec-donut/ec-donut.vue

```diff
@@ -73,47 +73,37 @@
   </div>
 </template>
 
-<script>
+<script setup>
+import { computed } from 'vue';
+
 import EcIcon from '../ec-icon';
 
-export default {
-  name: 'EcDonut',
-  components: { EcIcon },
-  props: {
-    amount: {
-      type: Number,
-      required: true,
-    },
-    used: {
-      type: Number,
-      required: true,
-    },
-  },
-  data() {
-    return {
-      radius: 48,
-    };
+const props = defineProps({
+  amount: {
+    type: Number,
+    required: true,
   },
-  computed: {
-    percentageUsed() {
-      if (this.used > this.amount) {
-        return 100;
-      }
-      if (this.used <= 0) {
-        return 0;
-      }
-      return (this.used / this.amount) * 100;
-    },
-    // Calculate the circumference
-    dashArray() {
-      return 2 * Math.PI * this.radius;
-    },
-    // Calculate the how much needs to be offset of the used circle
-    dashOffset() {
-      return this.dashArray * (1 - this.percentageUsed / 100);
-    },
+  used: {
+    type: Number,
+    required: true,
   },
-};
+});
+
+const radius = 48;
```

Comment:
In previous implementation this was a reactive parameter and now it isn't, why? If it is a contant then it should be naming in capslock case(RADIUS), so we could understand at the glance that it is unchangable value,

Reply:
thank you, I missed it, will change it to reactive param.

Reply:
But it never changes in the code, so maybe it is more constant than reactive variable?

Reply:
true, just realised it doesn't. Will edit the naming

Reply:
In Vue 2 there weren't many options to make variables non-reactive because everything attached to the data (the state of the component) was made reactive automatically. In Vue 3, we can control that and only have reactive things that are going to change in the future. Making the variable static (non-reactive) means that Vue 3 reactivity can skip lots of checks when comparing changes in VDOM a make the updates in DOM a little bit faster.

Agree, it should be renamed to RADIUS, no `ref()` needed.